lib/Request.js

```diff
@@ -48,9 +49,16 @@ Request.prototype.setUrl = function(url) {
         return value;
     });
 
+    //Remove props for query parameters
+    this.deleteQuery('props');
+
     this._etag = null;
 };
 
+Request.prototype.getValidationErrors = function() {
+    return this._validationErrors;
```

Comment:
But it's public, you don't get to decide when it's called. It's available for calling immediately after construction, which would not produce the behavior your'e expecting. I suggest making this private and validating automatically when calling the setters.